src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -0,0 +1,291 @@
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchTextureUsages(Workspace workspace, File texture, TextureType type) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			if (anyValueMatches(me.getGeneratableElement(), String.class, e -> {
+				TextureReference ref = e.getAnnotation(TextureReference.class);
+				return ref != null && ref.value() == type;
+			}, (a, t) -> {
+				TextureReference ref = a.getAnnotation(TextureReference.class);
+				return !List.of(ref.defaultValues()).contains(t) && workspace.getFolderManager()
+						.getTextureFile(FilenameUtilsPatched.removeExtension(t), type).equals(texture);
+			})) {
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchModelUsages(Workspace workspace, Model model) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			if (anyValueMatches(me.getGeneratableElement(), Model.class,
+					e -> e.isAnnotationPresent(ModelReference.class),
+					(a, t) -> model.equals(t) || TexturedModel.getModelTextureMapVariations(model).contains(t))) {
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchSoundUsages(Workspace workspace, SoundElement sound) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			if (anyValueMatches(me.getGeneratableElement(), Sound.class, e -> true,
+					(a, t) -> t.getUnmappedValue().replaceFirst("CUSTOM:", "").equals(sound.getName())))
+				elements.add(me);
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchStructureUsages(Workspace workspace, String structure) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			GeneratableElement ge = me.getGeneratableElement();
+			if (anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(StructureReference.class),
+					(a, t) -> t.equals(structure)))
+				elements.add(me);
+			else if (anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(BlocklyXML.class),
+					(a, t) -> t.contains(structure)))
+				elements.add(me);
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchGlobalVariableUsages(Workspace workspace, String variableName) {
+		List<ModElement> elements = new ArrayList<>();
+
+		workspace.getModElements().parallelStream().forEach(me -> {
+			if (anyValueMatches(me.getGeneratableElement(), String.class, e -> e.isAnnotationPresent(BlocklyXML.class),
+					(a, t) -> t.contains("<field name=\"VAR\">global:" + variableName + "</field>"))) {
+				elements.add(me);
+			}
+		});
+
+		return elements;
+	}
+
+	public static List<ModElement> searchLocalizationKeyUsages(Workspace workspace, String localizationKey) {

```

Comment:
Localization keys can also be used in procedures such as translated text or deal custom damage

Reply:
Isn't this the current behavior?
https://github.com/MCreator/MCreator/blob/df0a48b437116d3ca346f4cd09700b9e39eaff32/src/main/java/net/mcreator/workspace/references/ReferencesFinder.java#L157

Reply:
Not sure about the second case because I'd need to format block's message string using fields' values which sounds overcomplicated even without thinking about how to do this.

Reply:
I believe you can hardcode check for this edge-case

Reply:
I can try, but you'd consider the result an overkill.

Reply:
In this case, I prefer not to, I can try implementing something later as I believe no overkill would be needed

Reply:
:heavy_check_mark: Fixed